design/documentation-improvements.md

```diff
@@ -0,0 +1,128 @@
+* These modules are than assembled in _assemblies_ which create the 4 different documentation books.
+* All these modules are currently stored only in few directories
+
+Across the documentation, different variables / placeholders are used which allow easily substituting.
+These are used in place of things which are often changing such as versions.
+But also on other places - for example instead of saying just `Strimzi`, the documentation uses the variable `{ProductName}`.
+
+## Challenges
+
+The current documentation as described above has many challenges and is not friendly to contributors.
+As a result, most of the documentation work is done only by 2 people (@PaulRMellor and @laidan6000).
+While they do a great job, this situation is not optimal and it would be much better to have:
+
+* Bigger number of documentation contributors to better spread the load
+* Have the documentation work directly part of the PR which is changing a feature or adding new feature instead of having it done asynchronously by someone else
+
+This should also improve the release cycle which often has to wait for documentation for already merged features.
+With the documentation being done together with the code, the documentation will be either merged with the feature or the feature will not be merged.
+
+Apart form the way it influences the release cycle, the current documentation has also many other aspects which are not community friendly and which might discourage contributors:
+
+### It is hard to write
+
+* No tooling exists for working with the modular documentation.
+Contributor needs to find and edit the right files among several hundred of modules.
+There is no tool which would display the whole book and let the user to do changes to the individual modules from it.
+
+* It is hard to see how the changes / modules fit together because to see them all combined you need to render the documentation using Asciidoctor, open it in browser, find the right place, etc. 
+It is not just hard to find the files, the overall structure of the document is also very unclear from the edited files which makes it really hard to implement and changes or add new parts. 
+Even a simple change of few words feels like a chore.
+
+* Including single module many times is tempting and saves some effort. 
+But it also makes it harder to update because it is not clear on how many different places this change will be actually done.
+
+* The use of placeholders and variables for things such as the project name makes it hard to write the documentation since you need to know about these variables and keep them in mind all the time instead of focusing on the real content.
+
+### It is hard to review
+
+* It is not clear how the final document will look like and what will be the structure or order of the modules. 
+Since there is currently no CI/CD system available, authors often have to provide rendered version of the document “out of band”.
```

Comment:
```suggestion
Since there is currently no CI/CD system available, authors often have to provide a rendered version of the document “out of band”.
```